### PR TITLE
Report Creation API

### PR DESCRIPTION
**Pull request by piyushbeli**

### CODE REVIEW

File: src/services/report.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+let instance;
+
+class ResportService {
+    constructor(config, options) {
+        this._config = config;
+        this._logger = options['logger'];
+        this._db = options['db'];
+        this._Report = this._db[options.MODEL_NAME.REPORT];
```
Comment (piyushbeli) [inline src/services/report.js]: Can u please tell me the source of this data "options.MODEL_NAME.REPORT"? Where have we defined these constants how are we populating those in the "options".

Comment (piyushbeli) [reply src/services/report.js]: Can you also tell me what this._Report is holding? At LN:18 we are calling "create" method on this object so you can get an idea from there. If you will follow the code from "driver.js" then you should be able to answer this question.

Comment (meenalaggarwal) [reply src/services/report.js]: options.MODEL_NAME.REPORT is mapped from constants in utils.
And create function is used from the sequelize module which is initiated in models index.js which is called from driver.js file.

File: src/routes/report.js [hunk 2]
```diff
@@ -13,20 +14,38 @@ class ReportRoute extends iROUTER {
...
+        const {criteria} = req.body;
+        try {
+            if (!criteria) {
+                throw 'Required report creation param missing';
+            }
+            await this._service.createReport(req.body);
+            this.writeResponse(req, res);
+        } catch (e) {
+            this._logger.error({
+                msg: 'AuthRoute:signup:: Error occurred',
```
[the remark this answers is not in the document]
Comment (meenalaggarwal) [reply src/routes/report.js]: Ahh i missed that, i will keep that in mind.
Adding a commit for the same in branch

File: src/models/report.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+'use strict';
+module.exports = (sequelize, DataTypes) => {
+    const Report = sequelize.define('Report', {
+        criteria: DataTypes.STRING
```
[the remark this answers is not in the document]
Comment (meenalaggarwal) [reply src/models/report.js]: I think it can be either VARCHAR or STRING

File: src/routes/report.js [hunk 4]
```diff
@@ -13,20 +15,71 @@ class ReportRoute extends iROUTER {
         this._config = config;
         this._options = options;
         this._logger = options['logger'];
+        this._service = null;
+        this._schema = JOI.object().keys({
```
Comment (piyushbeli) [inline src/routes/report.js]: @meenalaggarwal , schema is the part of business logic and it should go into the service. Routes should be used only for extracting the data from the request and do some basic validations.

File: src/services/report.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+let instance;
+
+class ResportService {
```
Comment (piyushbeli) [inline src/services/report.js]: Typo mistake in the class name.

File: src/routes/report.js (hunk 4 above)
Comment (meenalaggarwal) [reply src/routes/report.js]: Updated

File: src/services/report.js (hunk 5 above)
Comment (meenalaggarwal) [reply src/services/report.js]: Updated

**Pull request merged**